logger.js

```diff
@@ -0,0 +1,205 @@
+const levels = {
+  info: "INFO",
+  warn: "WARN",
+  error: "ERROR",
+  debug: "DEBUG"
+}
+
+var currentLogLevel = "info"
+let isLoggerEnabled = true
+
+function getTimeStamp(){
+  const date = new Date();
+  return date.getFullYear() + "-" + (date.getMonth() + 1) + "-" + date.getDate() +
+      " " + date.getHours() + ":" + date.getMinutes() + ":" + date.getSeconds()
+}
+
+function log(level, message, details) {
+  if(!isLoggerEnabled) return;
+  let timeStamp = getTimeStamp();
+  let logEntry = `[${timeStamp}] [${level}] - ${message}`
+  if (details) { logEntry += " - Details: " + JSON.stringify(details) }
+  console.log(logEntry)
+  saveLogToFile(logEntry)
+}
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Consider log level check in logging</b></div>
<div id="fix">

The `log` function may benefit from checking if `level` matches `currentLogLevel` before proceeding with logging. Currently logs of all levels are being processed regardless of the set log level.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -17,6 +17,7 @@
 function log(level, message, details) {
   if(!isLoggerEnabled) return;
 + if(!shouldProcessLogLevel(level)) return;
   let timeStamp = getTimeStamp();
   let logEntry = `[${timeStamp}] [${level}] - ${message}`
   if (details) { logEntry += " - Details: " + JSON.stringify(details) }
```
</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/aakashgusain1990/BitoPythonTest/pull/21#issuecomment-2440543133>#378142</a></i></small>
</div>

Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged